echopype/convert/set_groups_azfp.py

```diff
@@ -167,18 +168,21 @@ def set_platform(self) -> xr.Dataset:
         """Set the Platform group."""
         platform_dict = {"platform_name": "", "platform_type": "", "platform_code_ICES": ""}
         unpacked_data = self.parser_obj.unpacked_data
+        time2 = self.parser_obj.ping_time
+        if np.isnan(unpacked_data["pressure"]).all():
+            depth = [np.nan] * len(time2)
+            water_level = [np.nan]
+        else:
+            depth = -1 * depth_from_pressure(
+                unpacked_data["pressure"], latitude=30, atm_pres_surf=0
+            )
+            water_level = [0.0]
 
         # Create nan time coordinate for lat/lon (lat/lon do not exist in AZFP 01A data)
         time1 = [np.nan]
 
-        # If tilt_x and/or tilt_y are all nan, create single-value time2 dimension
-        # and single-value (np.nan) tilt_x and tilt_y
-        tilt_x = [np.nan] if np.isnan(unpacked_data["tilt_x"]).all() else unpacked_data["tilt_x"]
-        tilt_y = [np.nan] if np.isnan(unpacked_data["tilt_y"]).all() else unpacked_data["tilt_y"]
-        if (len(tilt_x) == 1 and np.isnan(tilt_x)) and (len(tilt_y) == 1 and np.isnan(tilt_y)):
-            time2 = [self.parser_obj.ping_time[0]]
-        else:
-            time2 = self.parser_obj.ping_time
+        tilt_x = unpacked_data["tilt_x"]
+        tilt_y = unpacked_data["tilt_y"]
```

Comment:
You removed the block that checks if `unpacked_data["tilt_x"]` and `unpacked_data["tilt_y"]` are arrays with all nan, sets them to be length-1 nan arrays, and creates `time2` to be a length-1 array. But let's examine the possible scenarios:

tilt_x & tilt_y | vertical_offset | results for tilt_x, tilt_y, and vertical_offset
---- | ---- | ---- 
valid data | valid data | "full-length" arrays with valid data, with an unmodified `time2` dimension
valid data | invalid data (all nan) | full-length arrays where tilt_x & tilt_y have valid data and vertical_offset is all nan
invalid data (all nan) | valid data | full-length arrays where tilt_x & tilt_y are all nan and vertical_offset has valid data
invalid data (all nan) | invalid data (all nan) | length-1 arrays set to nan values, where `time2` dimension is set to `[self.parser_obj.ping_time[0]]`

For the 4th scenario (all invalid, nan data), I don't think your simplification would result in length-1 arrays for `time2`, `tilt_x`, `tilt_y`, and `vertical_offset`. Or does it?

Reply:
> For the 4th scenario (all invalid, nan data), I don't think your simplification would result in length-1 arrays for `time2`, `tilt_x`, `tilt_y`, and `vertical_offset`. Or does it?

Nope, but i thought even if `tilt_x`, `tilt_y` and `vertical_offset` are all `np.nan` we don't make them length - 1 arrays and instead keep them with their original dimension as it is. I think that should be modified?

Reply:
If `tilt_x`, `tilt_y` and `vertical_offset` are all nan, they don't really have an "original" dimension. But the SONAR-netCDF4 convention that we follow defines `vertical_offset` to have a time dimension, which in our case is `time2` (`tilt_x` and `tilt_y` are not defined by the convention). For AZFP files, no other variable in the Platform group that has valid data would use that dimension. So, setting `time2` to be length-1 reduces data storage, minimizes ambiguity, and allows us to maintain the time dimension in `vertical_offset`.

Reply:
thanks, got it. Updated the code!